character-sfx.js

```diff
@@ -97,6 +99,10 @@ class CharacterSfx {
       if (this.player.avatar.jumpState && !this.lastJumpState) {
         const audioSpec = soundFiles.jump[Math.floor(Math.random() * soundFiles.jump.length)];
         sounds.playSound(audioSpec);
+        // play jump grunt 
+        if(this.player.voicePack!==null && !this.player.hasAction('fly') && this.player.avatar.velocity.y<0){

```

Comment:
I think a better condition for this is `jumping && !lastJumping`.

Reply:
`jumping = this.player.hasAction('jump')`

Reply:
Instead of checking `this.player.voicePack!==null` I think it's better to let the `playGrunt` handle it.

Otherwise every other `playGrunt` call needs this check too (and it currently doesn't so I'm assuming it is bugged).

Reply:
Hm ok, but I still think it should check jump in addition to this. Otherwise if the user is hit and pushed back as a result, setting velocity up but not jump, the grunt will be played (but that would be wrong).

Reply:
Maybe we should add extra metadata to the `jump` action, like an extra key `trigger`:

`trigger === 'jump'` means they pressed the jump button
`trigger === 'fall'` means they got to jump mode because they fell off a cliff (and don't play the grunt in that case)

Reply:
For that solution we just need to add the extra metadata whenever we add the `jump` action.

Reply:
Hi @avaer so now I just add the `'jump' trigger` when player press the jump button in `ioManager`. https://github.com/webaverse/app/pull/2835/commits/8432ba8eff814b980e83a1e6dbbaab34d6559452

but I don't know where should I add the `'fall' trigger`. Should I add it in the z.mjs?
![unknown22](https://user-images.githubusercontent.com/60634884/164562769-3f85674e-d2a8-477c-886a-af6f291712db.png)